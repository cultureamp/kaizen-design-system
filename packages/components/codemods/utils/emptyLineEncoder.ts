// Fork of https://github.com/Serj-Tm/ts-empty-line-encoder/blob/master/encoder.ts
const DEFAULT_EMPTY_LINE_MARKER: string = '!--empty-line--!'
const DEFAULT_NEW_LINE: string = '\n'

const toComment = (marker: string): string => `/*${marker}*/`

export const encodeEmptyLines = (text: string): string => {
  const marker = toComment(DEFAULT_EMPTY_LINE_MARKER)
  const lines = text.split(/\r?\n/)
  const commentedLines = lines.map((line) => (line.trim() == '' ? marker : line))
  return commentedLines.join(DEFAULT_NEW_LINE)
}

export const decodeEmptyLines = (text: string): string => {
  const markerRegex = toComment(DEFAULT_EMPTY_LINE_MARKER).replace(/\*/g, '\\*')
  const decoded = text.replace(new RegExp(markerRegex, 'gi'), '')
  return decoded
}
