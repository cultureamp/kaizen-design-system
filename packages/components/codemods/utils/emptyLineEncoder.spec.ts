import fs from 'fs'
import path from 'path'
import { decodeEmptyLines, encodeEmptyLines } from './emptyLineEncoder'

describe('encodeEmptyLines()', () => {
  it('replaces empty lines with default line marker', () => {
    const filePath = path.resolve(path.join(__dirname, './__fixtures__/KaioComponent.tsx'))
    const fileContent = fs.readFileSync(filePath, 'utf8')
    const output = encodeEmptyLines(fileContent)
    expect(output).toMatchSnapshot()
  })
})

describe('decodeEmptyLines()', () => {
  it('replaces default line marker with empty line', () => {
    const filePath = path.resolve(path.join(__dirname, './__fixtures__/KaioComponent.tsx'))
    const fileContent = fs.readFileSync(filePath, 'utf8')
    const output = decodeEmptyLines(fileContent)
    expect(output).toMatchSnapshot()
  })
})
