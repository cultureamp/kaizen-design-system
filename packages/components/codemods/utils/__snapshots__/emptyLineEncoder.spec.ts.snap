// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`decodeEmptyLines() > replaces default line marker with empty line 1`] = `
"import React from 'react'
// @ts-ignore
import { Pancakes } from '@kaizen/components'

export const KaioComponent = (): JSX.Element => <Pancakes topping="butter" />
"
`;

exports[`encodeEmptyLines() > replaces empty lines with default line marker 1`] = `
"import React from 'react'
// @ts-ignore
import { Pancakes } from '@kaizen/components'
/*!--empty-line--!*/
export const KaioComponent = (): JSX.Element => <Pancakes topping="butter" />
/*!--empty-line--!*/"
`;
